boa/src/exec/mod.rs

```diff
@@ -256,17 +263,17 @@ impl Interpreter {
     pub(crate) fn to_primitive(
         &mut self,
         input: &mut Value,
-        preferred_type: Option<&str>,
+        preferred_type: Option<PreferredType>,
```

Comment:
I think we should take in just `PreferredType` instead of `Option<PreferredType>` because when it is `None` it's `"default"`